plugins/database/mongodb/mongodb.go

```diff
@@ -251,7 +279,7 @@ func (m *MongoDB) RevokeUser(ctx context.Context, statements dbplugin.Statements
 	err = session.DB(db).RemoveUser(username)
 	switch {
 	case err == nil, err == mgo.ErrNotFound:
-	case err == io.EOF, strings.Contains(err.Error(), "EOF"):
+	case err == io.EOF, strings.Contains(err.Error(), "EOF"), strings.Contains(err.Error(), "not master"):
 		if err := m.Close(); err != nil {
```

Comment:
I'm surprised we've not bumped into a deadlock here, but we should remove this `Close` call.

Reply:
Agreed. I removed it and replcaed it with a different way of closing that doesn't lead to deadlock.